mapper.go

```diff
@@ -285,6 +288,11 @@ func (r *Registry) RegisterDefaults() *Registry {
 		RegisterType(reflect.TypeOf(time.Duration(0)), durationDecoder()).
 		RegisterType(reflect.TypeOf(&url.URL{}), urlMapper()).
 		RegisterType(reflect.TypeOf(&os.File{}), fileMapper(r)).
+		RegisterType(reflect.TypeOf(&regexp.Regexp{}), regexMapper()).
+		RegisterType(reflect.TypeOf(&net.IP{}), netIPMapper()).
+		RegisterType(reflect.TypeOf(&net.IPNet{}), netIPNetMapper()).
+		RegisterType(reflect.TypeOf(netip.Addr{}), netipAddrMapper()).
+		RegisterType(reflect.TypeOf(netip.Prefix{}), netipPrefixMapper()).
```

Comment:
sorry for the drive-by review, but are these two actually necessary? :thinking: 

they implement `enconding.TextUnmarshaller` and should work out of the box?

Reply:
you are right, they work out of the box

Reply:
i removed the netip mappers from the pr, and cleanup my tests.

I thought about also removing the net mappers, and only add the regex mapper:
- this would guide users to use new new netip package instead of the net package

What do you think about that?

Reply:
Good question. Maybe both? I don't think there's any harm there.